### PR TITLE
Instructions for upgrading golang version

### PR DESCRIPTION
**Pull request by gongmax**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup

/kind documentation

> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:
Add a template to follow for upgrading Golang version

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/golang_version_upgrade.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            - [ ] Run `go mod edit -go=<NEW_GOLANG_VERSION> && go mod tidy`
+            - [ ] In `examples/simple-genai-server/Makefile`, increase the version number by 1 for `simple-genai-game-server`
+
+        - [ ] In `examples/supertuxkart`
+            - [ ] Run `go mod edit -go=<NEW_GOLANG_VERSION> && go mod tidy`
+            - [ ] In `examples/supertuxkart/Makefile`, increase the version number by 1 for `supertuxkart-example`
+
+        - [ ] In `examples/xonotic`
+            - [ ] Run `go mod edit -go=<NEW_GOLANG_VERSION> && go mod tidy`
+            - [ ] In `examples/xonotic/Makefile`, increase the version number by 1 for `xonotic-example`
```
Comment (Kalaiselvi84) [inline .github/ISSUE_TEMPLATE/golang_version_upgrade.md:58]: We can do this:
1) Navigate to examples directory
2) Run `find . -name 'go.mod' -execdir go mod edit -go=<NEW_GOLANG_VERSION> \;`
3) Run `find . -name 'go.mod' -execdir go mod tidy \;`

WDYT?

Comment (Kalaiselvi84) [reply .github/ISSUE_TEMPLATE/golang_version_upgrade.md:58]: We have a script to increment the examples image in all the files[not only /examples] https://github.com/googleforgames/agones/blob/main/build/scripts/bump-image/main.go#L15
 
Navigate to `/build` and run `make bump-image IMAGENAME=<imageName> VERSION=<current-version>`

File: .github/ISSUE_TEMPLATE/golang_version_upgrade.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - [ ] `find examples \( -name Dockerfile -o -name Dockerfile.windows \) -exec sed -i 's/golang:[0-9]\+\.[0-9]\+\.[0-9]\+/golang:<NEW_GOLANG_VERSION>/g' {} \;`
+    
+- [ ] Update the example images tag. At `build` directory, run:
+    - [ ] `make bump-image IMAGENAME=autoscaler-webhook VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=crd-client VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=custom-controller VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=simple-game-server VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=simple-genai-game-server VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=supertuxkart-example VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=xonotic-example VERSION=<current-image-version>`
```
Comment (Kalaiselvi84) [inline .github/ISSUE_TEMPLATE/golang_version_upgrade.md:21]: `allocation-endpoint` is missing from the list. Should we make the instructions more generic?


 Navigate to the `build` directory and use the following command to update each image tag. Replace <image-name> with the specific image name and <current-image-version> with the current version number:
 
 `make bump-image IMAGENAME=<image-name> VERSION=<current-image-version>`

File: .github/ISSUE_TEMPLATE/golang_version_upgrade.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+- [ ] Create a PR for the above changes and send for review
+
+- [ ] After the above PR is approved, **before** merging it, run the following to generate and push the new example images:
+    - [ ] In `examples/autoscaler-webhook`, run: `make cloud-build`
+    - [ ] In `examples/crd-client`, run: `make cloud-build`
+    - [ ] In `examples/custom-controller`, run: `make cloud-build`
+    - [ ] In `examples/simple-game-server`, run: `make cloud-build`
+    - [ ] In `examples/simple-genai-server`, run: `make cloud-build`
+    - [ ] In `examples/supertuxkart`, run: `make cloud-build`
+    - [ ] In `examples/xonotic`, run: `make cloud-build`
```
Comment (Kalaiselvi84) [inline .github/ISSUE_TEMPLATE/golang_version_upgrade.md:32]: same here, shall we make it generic?

File: .github/ISSUE_TEMPLATE/golang_version_upgrade.md (hunk 2 above)
Comment (gongmax) [reply .github/ISSUE_TEMPLATE/golang_version_upgrade.md:21]: I prefer to have more detailed instructions so someone who are not familiar with the code can still follow the steps.
Added instruction for `allocation-endpoint-proxy`

Comment (Kalaiselvi84) [reply .github/ISSUE_TEMPLATE/golang_version_upgrade.md:22]: > someone who are not familiar with the code can still follow the steps.

That's a good point! Just a thought...  shall we include these steps in a .sh file to automate the upgrade process? But the final call is yours....

```
#!/usr/bin/env bash

read -p "Enter the Golang version : " GOLANG_VERSION

echo "Updating go.mod files..."
find . -name 'go.mod' -not -path '*/\.*' -execdir go mod edit -go=$GOLANG_VERSION \;
find . -name 'go.mod' -not -path '*/\.*' -execdir go mod tidy \;

echo "Updating Dockerfiles in the build directory..."
find build -type f \( -not -path '*/\.*' -and -not -path 'build/tmp/*' \) -exec sed -i "s/GO_VERSION=[0-9]\+\.[0-9]\+\.[0-9]\+/GO_VERSION=$GOLANG_VERSION/g" {} \;

echo "Updating Dockerfiles in the examples directory..."
find examples -name Dockerfile -exec sed -i "s/golang:[0-9]\+\.[0-9]\+-alpine/golang:$GOLANG_VERSION-alpine/g" {} \;
find examples \( -name Dockerfile -o -name Dockerfile.windows \) -exec sed -i "s/golang:[0-9]\+\.[0-9]\+\.[0-9]\+/golang:$GOLANG_VERSION/g" {} \;

cd build

echo "Updating example images tags..."
read -p "Enter the current allocation-endpoint-proxy version: " VERSION_ALLOCATION_ENDPOINT_PROXY
make bump-image IMAGENAME=allocation-endpoint-proxy VERSION=$VERSION_ALLOCATION_ENDPOINT_PROXY

read -p "Enter the current autoscaler-webhook version: " VERSION_AUTOSCALER_WEBHOOK
make bump-image IMAGENAME=autoscaler-webhook VERSION=$VERSION_AUTOSCALER_WEBHOOK

read -p "Enter the current crd-client version: " VERSION_CRD_CLIENT
make bump-image IMAGENAME=crd-client VERSION=$VERSION_CRD_CLIENT

read -p "Enter the current custom-controller version: " VERSION_CUSTOM_CONTROLLER
make bump-image IMAGENAME=custom-controller VERSION=$VERSION_CUSTOM_CONTROLLER

read -p "Enter the current simple-game-server version: " VERSION_SIMPLE_GAME_SERVER
make bump-image IMAGENAME=simple-game-server VERSION=$VERSION_SIMPLE_GAME_SERVER

read -p "Enter the current simple-genai-game-server version: " VERSION_SIMPLE_GENAI_GAME_SERVER
make bump-image IMAGENAME=simple-genai-game-server VERSION=$VERSION_SIMPLE_GENAI_GAME_SERVER

read -p "Enter the current supertuxkart-example version: " VERSION_SUPERTUXKART_EXAMPLE
make bump-image IMAGENAME=supertuxkart-example VERSION=$VERSION_SUPERTUXKART_EXAMPLE

read -p "Enter the current xonotic-example version: " VERSION_XONOTIC_EXAMPLE
make bump-image IMAGENAME=xonotic-example VERSION=$VERSION_XONOTIC_EXAMPLE

```

Comment (zmerlynn) [reply .github/ISSUE_TEMPLATE/golang_version_upgrade.md:22]: I think given @gongmax's transition, we should just proceed with the checklist - we can automate this in the future, perhaps after the next bump (which will also help validate the checklist). Does that make sense?

**Review by markmandel — APPROVED** [review]

**Pull request merged**